services/core-web/common/models/documents/document.ts

```diff
@@ -0,0 +1,153 @@
+import { USER_ROLES } from "@mds/common";
+
+export enum FileOperations {
+  View = "Open in document viewer",
+  Download = "Download file",
+  Replace = "Replace file",
+  Archive = "Archive file",
+  Delete = "Delete",
+}
+
+/* 
+A base class for Mine Documents
+
+There is an issue with antd where sorting a table that has children (ie matchChildColumnsToParent)
+will transform the records into type <any> (with versions still maintaining their type) so properties accessed
+by the table should be *set* to the specific object, cannot expect to be able to consistently call its methods
+
+include "user_roles" property in the json used in the constructor to set allowed actions based on the user
+*/
+export class MineDocument {
+  public mine_document_guid: string;
+
+  public mine_guid: string;
+
+  public document_manager_guid: string;
```

Comment:
eslint added all these spaces on me and I was not feeling like changing eslint rules, even though it irritates me to look at it.